queries/nu/highlights.scm

```diff
@@ -0,0 +1,396 @@
+; ---
+; keywords
+[
+  "def"
+  "alias"
+  "export-env"
+  "export"
+  "extern"
+  "module"
+  "let"
+  "let-env"
+  "mut"
+  "const"
+  "hide-env"
+  "source"
+  "source-env"
+  "overlay"
+  "register"
+  "loop"
+  "while"
+  "error"
+  "do"
+  "if"
+  "else"
+  "try"
+  "catch"
+  "match"
+  "break"
+  "continue"
+  "return"
+] @keyword
+
+(hide_mod
+  "hide" @keyword)
+
+(decl_use
+  "use" @keyword)
+
+(ctrl_for
+  "for" @keyword
+  "in" @keyword)
+
+(overlay_list
+  "list" @keyword.import)
+
+(overlay_hide
+  "hide" @keyword.import)
+
+(overlay_new
+  "new" @keyword.import)
+
+(overlay_use
+  "use" @keyword.import
+  "as" @keyword)
+
+(ctrl_error
+  "make" @keyword.import)
+
+; ---
+; literals
+(val_number) @number
+
+(val_duration
+  unit: _ @variable.parameter)
+
+(val_filesize
+  unit: _ @variable.parameter)
+
+(val_binary
+  [
+    "0b"
+    "0o"
+    "0x"
+  ] @number
+  "[" @punctuation.bracket
+  digit: [
+    "," @punctuation.delimiter
+    (hex_digit) @number
+  ]
+  "]" @punctuation.bracket) @number
+
+(val_bool) @constant.builtin
+
+(val_nothing) @constant.builtin
+
+(val_string) @string
+
+arg_str: (val_string) @variable.parameter
+
+file_path: (val_string) @variable.parameter
+
+(val_date) @number
+
+(inter_escape_sequence) @string.escape
+
+(escape_sequence) @string.escape
+
+(val_interpolated
+  [
+    "$\""
+    "$\'"
+    "\""
+    "\'"
+  ] @string)
+
+(unescaped_interpolated_content) @string
+
+(escaped_interpolated_content) @string
+
+(expr_interpolated
+  [
+    "("
+    ")"
+  ] @variable.parameter)
+
+; ---
+; operators
+(expr_binary
+  [
+    "+"
+    "-"
+    "*"
+    "/"
+    "mod"
+    "//"
+    "++"
+    "**"
+    "=="
+    "!="
+    "<"
+    "<="
+    ">"
+    ">="
+    "=~"
+    "!~"
+    "and"
+    "or"
+    "xor"
+    "bit-or"
+    "bit-xor"
+    "bit-and"
+    "bit-shl"
+    "bit-shr"
+    "in"
+    "not-in"
+    "starts-with"
+    "ends-with"
+  ] @operator)
```

Comment:
Would it be valid to merge all of these operator highlights just under a top-level query (no where_command/ expr_binary parent captures)? and same with the other operator highlights? Seems unnecessary to only highlight them under these parent groups, unless there is a good reason I'm missing

Reply:
Do not have any strong opinion on this since I am new to both tree-sitter grammer and nu. I copied this code from the official nushell tree-sitter repo and if you feel it's better to merge these I will take your word on it